nodebuilder/da/service.go

```diff
@@ -126,6 +152,36 @@ func (s *Service) Submit(
 	return ids, nil
 }
 
+func parseOptions(options []byte) ([]state.ConfigOption, error) {
+	var opts []state.ConfigOption
+
+	parsedOpts := struct {
+		KeyName           string
+		SignerAddress     string
+		FeeGranterAddress string
+	}{}
```

Comment:
This options are implementation specific. It doesn't even have to be JSON in other implementations of go-da.
It's also up to implementation to specify supported options, which can be very different from the ones defined in celestia-node. That's why we don't define them in go-da.

We don't care about encoding to JSON as here we only parse (using reasonable defaults) - rollup operator will prepare actual value of `options` as a part of a rollkit config or start script.

Reply:
Does it imply, that client should be aware of implementation specific options when using generic go-da interface? 
There could be few problems with this approach:
- different name for same options in different implementations
- client will need to depend on implementation specific encoding of options instead of interface definition only.

If go-da does not define encoding, we should define it in node as separate struct with json annotation tags for fields. Otherwise other clients (non go-da) won't be able to properly encode options.

Reply:
Let's clarify entire setup a bit.

1. Rollkit is a client of go-da API. Rollkit is DA implementation agnostic.
2. celestia-node is one of go-da implementations, acting as server of go-da API.
3. Each time a user configures rollkit to run a rollup, he needs to setup DA - he knows:
  - the type of DA (celestia-node, avail, bitcoin, etc)
  - endpoint address
  - options to use (for example: key name for this concrete instance of celestia-node that he wish to use) 

Typical real-world usage for `SubmitWithOption` is RaaS setup - RaaS provider runs a single celestia-node with multiple keys in keyring. Rollkit rollup is then configured to pass `{"KeyName": "value agreed with RaaS provider"}` as `--rollkit.da_submit_options`. 
Those options are not interpreted or parsed by Rollkit, it's just passed to go-da implementation - any parsing errors are returned by go-da server/implementation.

This change only applies to go-da implementation, so I don't expect "other clients".  But I agree that this should be documented somewhere - this will be definitely a part of rollkit doc, but if there is a good place to document this in celestia-node please let me know.

I can also refactor so the struct is exported and properly described in godoc.

Reply:
I'd prefer the latter (export + describe in godoc).